### PR TITLE
Sync with neo and add default values

### PR DESCRIPTION
**Pull request by chenzhitong**

### CODE REVIEW

**Comment by shargon** [comment]
If we will add default values, we should review if we can add more of them

**Comment by erikzhang** [comment]
BTW, the NFT contract should be updated too. (Added new method `tokens()`.)

**Review by shargon — APPROVED** [review]

File: src/Neo.SmartContract.Framework/Services/Neo/ContractManagement.cs [hunk 1]
```diff
@@ -7,8 +7,8 @@ public class ContractManagement
     {
         public static extern UInt160 Hash { [ContractHash] get; }
         public static extern Contract GetContract(UInt160 hash);
-        public static extern Contract Deploy(ByteString nefFile, string manifest, object data);
-        public static extern void Update(ByteString nefFile, string manifest, object data);
+        public static extern Contract Deploy(ByteString nefFile, string manifest, object data = null);
+        public static extern void Update(ByteString nefFile, string manifest, object data = null);
```
Comment (erikzhang) [inline src/Neo.SmartContract.Framework/Services/Neo/ContractManagement.cs:11]: They have overloads.

Comment (shargon) [reply src/Neo.SmartContract.Framework/Services/Neo/ContractManagement.cs]: Done

**Review by erikzhang — DISMISSED** [review]

**Comment by shargon** [comment]
Require https://github.com/neo-project/neo/pull/2277

**Review by vncoelho — APPROVED** [review]

File: templates/Template.CSharp/Contract1.cs [hunk 2]
```diff
@@ -3,12 +3,14 @@
...
 using System;
 using System.Numerics;
+using System.ComponentModel;
 
 namespace $safeprojectname$
 {
     [ManifestExtra("Author", "Neo")]
     [ManifestExtra("Email", "dev@neo.org")]
     [ManifestExtra("Description", "This is a contract example")]
+    [DisplayName("Name")]
```
Comment (erikzhang) [inline templates/Template.CSharp/Contract1.cs:13]: Why do we have to add a `DisplayName`? We can use the class name as the contract name.

Comment (shargon) [reply templates/Template.CSharp/Contract1.cs:13]: Yes, By default we can do it, I will change it

Comment (shargon) [reply templates/Template.CSharp/Contract1.cs]: Done

**Review by shargon — APPROVED** [review]

File: src/Neo.Compiler.MSIL/FuncExport.cs [hunk 3]
```diff
@@ -202,7 +201,7 @@ public static string GenerateManifest(JObject abi, NeoModule module)
             var name = module.attributes
```
Comment (erikzhang) [inline src/Neo.Compiler.MSIL/FuncExport.cs:201]: I think we should use the class name because the attribute is applied on the class.

Comment (shargon) [reply src/Neo.Compiler.MSIL/FuncExport.cs:201]: Got it. It will use now the class name if there are only one, otherwise, it will use the filename.

File: tests/Neo.Compiler.MSIL.UnitTests/TestClasses/Contract1.cs [hunk 4]
```diff
@@ -1,5 +1,8 @@
+using System.ComponentModel;
+
 namespace Neo.Compiler.MSIL.UnitTests.TestClasses
-{
+{
+    [DisplayName("Name")]
```
Comment (erikzhang) [inline tests/Neo.Compiler.MSIL.UnitTests/TestClasses/Contract1.cs:5]: Now we can revert these changes?

Comment (shargon) [reply tests/Neo.Compiler.MSIL.UnitTests/TestClasses/Contract1.cs]: Done

**Review by shargon — APPROVED** [review]

**Review by vncoelho — APPROVED** [review]

File: src/Neo.Compiler.MSIL/FuncExport.cs (hunk 3 above)
Comment (erikzhang) [reply src/Neo.Compiler.MSIL/FuncExport.cs:201]: Why do we allow multiple contract classes in a project? If so, we should generate multiple `.nef` file.

**Comment by Tommo-L** [comment]
Merge?
